test/ERC6538Registry.t.sol

```diff
@@ -343,168 +320,58 @@ contract RegisterKeysOnBehalf is ERC6538RegistryTest {
     }
   }
 }
 
 contract Domain_Separator is ERC6538RegistryTest {
-  function testFuzz_ChecksTheDomainSeparatorOfTheErc6538RegistryContractAfterAChainSplitIsRecomputedWithTheNewChainId(
-    uint256 chainId
-  ) external {
-    chainId = bound(chainId, 1, 1_000_000);
-    vm.chainId(chainId);
-    assertEq(
-      registry.DOMAIN_SEPARATOR(),
-      keccak256(
-        abi.encode(
-          keccak256(
-            "EIP712Domain(string name,string version,uint256 chainId,address registryContract)"
-          ),
-          keccak256("ERC6538Registry"),
-          keccak256("1.0"),
-          chainId,
-          address(registry)
-        )
-      )
-    );
-  }
-}
-
-contract ERC1271MockContract {
-  address public owner;
-
-  enum RecoverError {
-    NoError,
-    InvalidSignature,
-    InvalidSignatureLength,
-    InvalidSignatureS
-  }
-
-  error ECDSAInvalidSignature();
-  error ECDSAInvalidSignatureLength(uint256 length);
-  error ECDSAInvalidSignatureS(bytes32 s);
-
-  constructor() {
-    owner = msg.sender;
+  function test_ReturnsTheValueComputedAtDeploymentIfChainIdRemainsUnchanged() external {
+    ERC6538RegistryHarness _registry = new ERC6538RegistryHarness();
+    assertEq(_registry.DOMAIN_SEPARATOR(), _registry.exposed_INITIAL_DOMAIN_SEPARATOR());
   }
 
-  function isValidSignature(bytes32 hash, bytes memory signature)
-    public
-    view
-    returns (bytes4 magicValue)
-  {
-    return recover(hash, signature) == owner ? this.isValidSignature.selector : bytes4(0);
-  }
-
-  function recover(bytes32 hash, bytes memory signature) internal pure returns (address) {
-    (address recovered, RecoverError error, bytes32 errorArg) = tryRecover(hash, signature);
-    _throwError(error, errorArg);
-    return recovered;
-  }
-
-  function tryRecover(bytes32 hash, bytes memory signature)
-    internal
-    pure
-    returns (address, RecoverError, bytes32)
-  {
-    if (signature.length == 65) {
-      bytes32 r;
-      bytes32 s;
-      uint8 v;
-      // ecrecover takes the signature parameters, and the only way to get them currently is to use
-      // assembly.
-      assembly ("memory-safe") {
-        r := mload(add(signature, 0x20))
-        s := mload(add(signature, 0x40))
-        v := byte(0, mload(add(signature, 0x60)))
-      }
-      return tryRecover(hash, v, r, s);
-    } else {
-      return (address(0), RecoverError.InvalidSignatureLength, bytes32(signature.length));
-    }
-  }
-
-  function tryRecover(bytes32 hash, uint8 v, bytes32 r, bytes32 s)
-    internal
-    pure
-    returns (address, RecoverError, bytes32)
+  function testFuzz_ReturnsARecomputedValueIfTheChainIdChanges(uint256 chainId1, uint256 chainId2)
+    external
   {
-    if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0) {
-      return (address(0), RecoverError.InvalidSignatureS, s);
-    }
+    chainId1 = bound(chainId1, 1, 2 ** 64 - 1);
+    chainId2 = bound(chainId2, 1, 2 ** 64 - 1);
```

Comment:
why `2 ** 64 - 1`?

Reply:
This is the highest supported chainId so there's no reason not to fuzz across the whole range.